src/ExpenseTracker.Infrastructure/Repositories/Implementation/UserRepository.cs

```diff
@@ -9,5 +10,8 @@ public class UserRepository : GenericRepository<User>, IUserRepository
         public UserRepository(DbContext context) : base(context)
         {
         }
+
+        public async Task<bool> ExistingUserName(string username)
```

Comment:
@BirajMainali I think the  method name could be more descriptive. Something like `IsUsernameUsed` or `UserExists(string username)` might be more readable.